pkg/chart/chart.go

```diff
@@ -394,6 +404,13 @@ func (c *OCIClient) InitClient(appRepo *appRepov1.AppRepository, caCertSecret *c
 			return err
 		}
 		headers.Set("Authorization", string(auth))
+	} else if registryCreds != nil && appRepo.Spec.Auth.RegistryCreds != nil {
```

Comment:
Just to be sure: there is no case in which both `authSecret` and `registryCreds` are being passed, no? If so, we should say somewhere that the latter takes precedence over the former (and add a test with both parameters).
If not, this ok as is.

Reply:
I have changed this to use the same secret and treat the case in which the secret key is `.dockerconfigjson` differently.